Artifacts/PgSql/Structure/Security/0050-Tables-MultipleAuthStrats.sql

```diff
@@ -0,0 +1,126 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+------------------------------ dbo.ResourceClaimActions -------------------------------------------------
+CREATE TABLE dbo.ResourceClaimActions(
+	ResourceClaimActionId SERIAL NOT NULL,
+	ActionId INT NOT NULL,
+	ResourceClaimId INT NOT NULL,
+	ValidationRuleSetName VARCHAR NULL,
+ CONSTRAINT ResourceClaimActionId_PK PRIMARY KEY
+(
+	ResourceClaimActionId 
+));
+
+CREATE INDEX IF NOT EXISTS IX_ResourceClaimActions_ActionId
+    ON dbo.ResourceClaimActions
+	(ActionId);
+
+CREATE INDEX IF NOT EXISTS IX_ResourceClaimActions_ResourceClaimId
+    ON dbo.ResourceClaimActions
+	(ResourceClaimId);
+
+ALTER TABLE dbo.ResourceClaimActions  ADD  CONSTRAINT FK_ResourceClaimActions_Actions FOREIGN KEY(ActionId)
+REFERENCES dbo.Actions (ActionId)
+ON DELETE CASCADE;
+
+ALTER TABLE dbo.ResourceClaimActions  ADD  CONSTRAINT FK_ResourceClaimActions_ResourceClaims FOREIGN KEY(ResourceClaimId)
+REFERENCES dbo.ResourceClaims (ResourceClaimId)
+ON DELETE CASCADE;
+
+---------------------------- dbo.ClaimSetResourceClaimActions --------------------------------------------------------
+
+CREATE TABLE dbo.ClaimSetResourceClaimActions(
+	ClaimSetResourceClaimActionId SERIAL NOT NULL,
+	ActionId INT NOT NULL,
+	ClaimSetId INT NOT NULL,
+	ResourceClaimId INT NOT NULL,
+	ValidationRuleSetNameOverride VARCHAR NULL,
+ CONSTRAINT ClaimSetResourceClaimActions_PK PRIMARY KEY  
+(
+	ClaimSetResourceClaimActionId 
+));
+
+CREATE INDEX IF NOT EXISTS IX_ClaimSetResourceClaimActions_ActionId
+    ON dbo.ClaimSetResourceClaimActions(ActionId);
+
+CREATE INDEX IF NOT EXISTS IX_ClaimSetResourceClaimActions_ClaimSetId
+    ON dbo.ClaimSetResourceClaimActions(ClaimSetId);
+
+CREATE INDEX IF NOT EXISTS IX_ClaimSetResourceClaimActions_ResourceClaimId
+    ON dbo.ClaimSetResourceClaimActions(ResourceClaimId);
+
+
+ALTER TABLE dbo.ClaimSetResourceClaimActions ADD CONSTRAINT FK_ClaimSetResourceClaimActions_Actions FOREIGN KEY(ActionId)
+REFERENCES dbo.Actions (ActionId)
+ON DELETE CASCADE;
```

Comment:
@vimayya Do we really want to cascade delete metadata if someone deletes an Action? This makes sense to me for ClaimSets, and even ResourceClaimId... but Actions, they should rarely, if ever, create or remove these.

@semalaiappan The answer to this question would need to be applied to SQL Server version of this script as well.

Reply:
I updated here ON DELETE CASCADE removed for actions table